applications/CompressiblePotentialFlowApplication/custom_elements/incompressible_perturbation_potential_flow_element.cpp

```diff
@@ -469,10 +468,42 @@ void IncompressiblePerturbationPotentialFlowElement<Dim, NumNodes>::CalculateLef
 
         CalculateLeftHandSideSubdividedElement(lhs_positive, lhs_negative, rCurrentProcessInfo);
         AssignLeftHandSideSubdividedElement(rLeftHandSideMatrix, lhs_positive,
-                                           lhs_negative, lhs_total, data);
+                                           lhs_negative, lhs_total, lhs_wake_condition, data);
     }
     else
-        AssignLeftHandSideWakeElement(rLeftHandSideMatrix, lhs_total, data);
+        AssignLeftHandSideWakeElement(rLeftHandSideMatrix, lhs_total, lhs_wake_condition, data);
+}
+
+// In 2D
+template <>
+void IncompressiblePerturbationPotentialFlowElement<2, 3>::CalculateBlockLeftHandSideWakeElement( BoundedMatrix<double, 3, 3>& rLhs_total, BoundedMatrix<double, 3, 3>& rLhs_wake_condition, const ElementalData<3, 2>& rData, const ProcessInfo& rCurrentProcessInfo)
+{
+    const double free_stream_density = rCurrentProcessInfo[FREE_STREAM_DENSITY];
+    ComputeLHSGaussPointContribution(rData.vol*free_stream_density, rLhs_total, rData);
+    rLhs_wake_condition = rLhs_total;
+}
+
+// In 3D
+template <>
+void IncompressiblePerturbationPotentialFlowElement<3, 4>::CalculateBlockLeftHandSideWakeElement( BoundedMatrix<double, 4, 4>& rLhs_total, BoundedMatrix<double, 4, 4>& rLhs_wake_condition, const ElementalData<4, 3>& rData, const ProcessInfo& rCurrentProcessInfo)
+{
+    const double free_stream_density = rCurrentProcessInfo[FREE_STREAM_DENSITY];
+    ComputeLHSGaussPointContribution(rData.vol*free_stream_density, rLhs_total, rData);
+
+    // Computing linearized pressure equality condition lhs
+    const array_1d<double, 3> free_stream_velocity_direction = rCurrentProcessInfo[FREE_STREAM_VELOCITY_DIRECTION];
```

Comment:
```suggestion
    const array_1d<double, 3>& free_stream_velocity_direction = rCurrentProcessInfo[FREE_STREAM_VELOCITY_DIRECTION];
```

Reply:
My suggestions were mainly motivated by what @philbucher suggested in a previous PR:

https://github.com/KratosMultiphysics/Kratos/pull/8432#discussion_r588138591

For doubles, my understanding is that the memory reduction doing this is neglible. Is there an issue accessing these values by reference as well @philbucher @rubenzorrilla ?